pkg/model/project.go

```diff
@@ -427,6 +404,59 @@ func (p *Project) GetAllUserGroups() []*ProjectUserGroup {
 	return all
 }
 
+// IsAssigned checks whether the team is already assigned as built-in roles.
+func (p *ProjectRBACConfig) IsAssigned(team string) bool {
+	if p == nil {
+		return false
+	}
+	return p.Admin == team || p.Editor == team || p.Viewer == team
+}
+
+// AddUserGroup adds a user group.
+func (p *Project) AddUserGroup(sso, role string) error {
+	if p.HasUserGroup(sso) {
+		return fmt.Errorf("%s is already assigned role. The SSO group must be unique", sso)
+	}
+	if p.Rbac.IsAssigned(sso) {
+		return fmt.Errorf("%s is already assigned role. The SSO group must be unique", sso)
+	}
+	if !p.HasRBACRole(role) {
+		return fmt.Errorf("%s role doesn't exist", role)
+	}
+	p.UserGroups = append(p.UserGroups, &ProjectUserGroup{
+		SsoGroup: sso,
+		Role:     role,
+	})
+	return nil
+}
+
+// DeleteUserGroup deletes a user group.
+func (p *Project) DeleteUserGroup(sso string) error {
+	for i, v := range p.UserGroups {
+		if v.SsoGroup == sso {
+			c := copy(p.UserGroups[i:], p.UserGroups[i+1:])
+			p.UserGroups = p.UserGroups[:i+c]
+			return nil
+		}
+	}
+	if p.Rbac != nil {
+		if p.Rbac.Admin == sso {
+			p.Rbac.Admin = ""
+			return nil
+		}
+		if p.Rbac.Editor == sso {
+			p.Rbac.Editor = ""
+			return nil
+		}
+		if p.Rbac.Viewer == sso {
+			p.Rbac.Viewer = ""
+			return nil
+		}
```

Comment:
If they are having the same value, shouldn't all of them be removed?
I mean maybe we don't need the `return`.

Reply:
True, Thank you🙇